addons/website_form/static/src/snippets/s_website_form/options.js

```diff
@@ -736,11 +738,20 @@ options.registry.WebsiteFieldEditor = FieldEditor.extend({
     willStart: async function () {
         const _super = this._super.bind(this);
         // Get the authorized existing fields for the form model
-        this.existingFields = await this._rpc({
-            model: "ir.model",
-            method: "get_authorized_fields",
-            args: [this.formEl.dataset.model_name],
-        }).then(fields => {
+        const model = this.formEl.dataset.model_name;
+        let fields;
+        if (model in authorized_fields_cache){
+            fields = authorized_fields_cache[model];
+        } else {
+            fields = await this._rpc({
+                model: "ir.model",
+                method: "get_authorized_fields",
+                args: [this.formEl.dataset.model_name],
+            });
```

Comment:
Same comment as for the image cache: your cache should hold the Promises instead of the final values so that it still work with the successive calls.